omxd.c

```diff
@@ -58,10 +96,24 @@ int main(int argc, char *argv[])
 	return 0;
 }
 
-/* Fork, umask, SID, chdir, close, logfile, FIFO */
+/* Open the FIFO at /var/run/omxctl */
+static int open_fifo(void)
+{
+	/* Create and open FIFO for command input as stdin */
+	unlink("omxctl");
+	LOG(1, "open_fifo: Deleted original omxctl FIFO.\n");
+	const mode_t old_mask = umask(0);
+	if (mknod("omxctl", S_IFIFO | 0622, 0) < 0)
+		return 6;
+	umask(old_mask);
+	LOG(1, "open_fifo: Created new omxctl FIFO.\n");
+
+	return 0;
+}
+
+/* Fork, umask, SID, close, logfile. */
 static int daemonize(void)
 {
-	I_root = geteuid() == 0;
```

Comment:
I see the need for the euid check coming out of daemonize, bit please keep I_root, I think it's used elsewhere too.